MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/AddressServiceDefault.java

```diff
@@ -48,46 +48,50 @@
 
     final Scheduler callbackScheduler;
 
+    final CustomerService customerService;
+
     AddressServiceDefault(
         final Retrofit retrofit,
         final NetworkRetryPolicyProvider networkRetryPolicyProvider,
-        final Scheduler callbackScheduler
+        final Scheduler callbackScheduler,
+        final CustomerService customerService
     ) {
         this.retrofitService = retrofit.create(AddressRetrofitService.class);
         this.networkRetryPolicyProvider = networkRetryPolicyProvider;
         this.callbackScheduler = callbackScheduler;
+        this.customerService = customerService;
     }
 
     @Override
-    public CancellableTask createAddress(final Long customerId, final Address address, final Callback<Address> callback) {
-        return new CancellableTaskSubscriptionWrapper(createAddress(customerId, address).subscribe(new InternalCallbackSubscriber<>(callback)));
+    public CancellableTask createAddress(final Address address, final Callback<Address> callback) {
+        return new CancellableTaskSubscriptionWrapper(createAddress(address).subscribe(new InternalCallbackSubscriber<>(callback)));
     }
 
     @Override
-    public Observable<Address> createAddress(final Long customerId, final Address address) {
-        if (customerId == null) {
-            throw new NullPointerException("customerId cannot be null");
+    public Observable<Address> createAddress(final Address address) {
+        if (customerService.getCustomerToken() == null) {
```

Comment:
We better extract this to local variable:
CustomerToken customerToken = customerService.getCustomerToken()

and use it later in createAddress to avoid possible race condition